src/protocol/ETHBridge.sol

```diff
@@ -62,15 +64,35 @@ contract ETHBridge is IETHBridge, ReentrancyGuardTransient {
 
     /// @inheritdoc IETHBridge
     function claimDeposit(ETHDeposit memory ethDeposit, uint256 height, bytes memory proof) external nonReentrant {
-        bytes32 id = _generateId(ethDeposit);
+        bytes32 id = _claimDeposit(ethDeposit, ethDeposit.to, ethDeposit.data, height, proof);
+        emit DepositClaimed(id, ethDeposit);
+    }
+
+    /// @inheritdoc IETHBridge
+    function cancelDeposit(ETHDeposit memory ethDeposit, address claimee, uint256 height, bytes memory proof)
```

Comment:
Missing nonReentrant, might be worth adding to the `__claimDeposit` function to have it both in `cancelDeposit` and `claimDeposit`